.github/workflows/build-rd-cpp.yml

```
@@ -0,0 +1,234 @@
+          endif()
+          file(STRINGS environment_script_output.txt output_lines)
+          foreach(line IN LISTS output_lines)
+            if (line MATCHES "^([a-zA-Z0-9_-]+)=(.*)$")
+              set(ENV{${CMAKE_MATCH_1}} "${CMAKE_MATCH_2}")
+            endif()
+          endforeach()
+        endif()
+
+        set(path_separator ":")
+        if ("${{ runner.os }}" STREQUAL "Windows")
+          set(path_separator ";")
+        endif()
+        set(ENV{PATH} "$ENV{GITHUB_WORKSPACE}${path_separator}$ENV{PATH}")
+
+        file(MAKE_DIRECTORY build)
+        execute_process(
+          COMMAND ${CMAKE_COMMAND}
+            -S .
+            -B build
+            -DENABLE_TESTS_OPTION:BOOL=OFF
+            -DCMAKE_BUILD_TYPE=$ENV{BUILD_TYPE}
+            -G Ninja
+            -DCMAKE_MAKE_PROGRAM=${CMAKE_CURRENT_SOURCE_DIR}/ninja
+          RESULT_VARIABLE build_result
+        )
+        if (NOT build_result EQUAL 0)
+          message(FATAL_ERROR "Bad exit status. Fail building RD CPP")
+        endif()
+
+
+    - name: Build
+      working-directory: ${{ env.working-dir }}
+      shell: cmake -P {0}
+      run: |
+        set(ENV{NINJA_STATUS} "[%f/%t %o/sec] ")
+        file(GLOB all_files "*")
+        message(${all_files})
+
+        if ("${{ runner.os }}" STREQUAL "Windows" AND NOT "x${{ matrix.config.environment_script }}" STREQUAL "x")
+          file(STRINGS environment_script_output.txt output_lines)
+          foreach(line IN LISTS output_lines)
+            if (line MATCHES "^([a-zA-Z0-9_-]+)=(.*)$")
+              set(ENV{${CMAKE_MATCH_1}} "${CMAKE_MATCH_2}")
+            endif()
+          endforeach()
+        endif()
+
+        set(path_separator ":")
+        if ("${{ runner.os }}" STREQUAL "Windows")
+          set(path_separator ";")
+        endif()
+        set(ENV{PATH} "$ENV{GITHUB_WORKSPACE}${path_separator}$ENV{PATH}")
+
+        execute_process(
+          COMMAND ${CMAKE_COMMAND} --build build --config ${{ env.BUILD_TYPE }}
+          RESULT_VARIABLE result
+        )
+        if (NOT result EQUAL 0)
+          message(FATAL_ERROR "Bad exit status")
+        endif()
+
+    - name: Install Strip
+      working-directory: ${{ env.working-dir }}
+      run: cd build && cmake --config ${{ env.BUILD_TYPE }} -P cmake_install.cmake
+    
+    - name: Pack
+      working-directory: ${{ env.working-dir }}
+      run: cd export && cmake -E tar "cfv" ../${{ env.artifact }} --format=zip include libs
+
+    - name: Upload
+      uses: actions/upload-artifact@v2
+      with:
+        path:  |
+          ${{ env.working-dir }}${{ matrix.config.SEP }}export${{ matrix.config.SEP }}**
+        name: ${{ env.artifact }}
+
+    - name: Pack Nuget
+      working-directory: ${{ env.working-dir }}${{ matrix.config.SEP }}export
+      run: nuget pack rd-cpp.nuspec -Version ${{ env.version }}
+
+    - name: Push nuget
+      uses: tanaka-takayoshi/nuget-publish-to-github-packages-action@v2.1
```

Currently, we'll push a package even for pull request, which shouldn't be the case. I suggest to change it as this:
```suggestion
      uses: tanaka-takayoshi/nuget-publish-to-github-packages-action@v2.1
      if: github.event_name == 'commit'
```

This should filter it to publish only the commits made on the master branch, and disable for pull requests.